metagraph/src/graph/graph_extensions/node_rc.cpp

```diff
@@ -0,0 +1,275 @@
+        } else {
+            std::string rev_seq = dbg_succ_->get_node_sequence(node).substr(1);
+            assert(rev_seq.size() == boss.get_k());
+
+            if (rev_seq[0] == BOSS::kSentinel)
+                return;
+
+            ::reverse_complement(rev_seq.begin(), rev_seq.end());
+            auto encoded = boss.encode(rev_seq);
+            auto [edge, edge_2, end] = boss.index_range(encoded.begin(), encoded.end());
+
+            if (end == encoded.end()) {
+                assert(edge == edge_2);
+                rc_edge = edge;
+            }
+        }
+
+        if (!rc_edge)
+            return;
+
+        // rc_edge may be a dummy sink, so this won't work with graph_->call_incoming_kmers
+        boss.call_incoming_to_target(boss.bwd(rc_edge), boss.get_node_last_value(rc_edge),
+            [&](edge_index incoming_boss_edge) {
+                node_index next = dbg_succ_->boss_to_kmer_index(incoming_boss_edge);
+                if (next != DeBruijnGraph::npos)
+                    callback(next);
+            }
+        );
+
+        return;
+    }
+
+    // Do the checks by directly mapping the sequences of the desired k-mers.
+    // For non-DBGSuccinct graphs, this should be fast enough.
+    std::string rev_seq = graph_->get_node_sequence(node).substr(1) + std::string(1, '\0');
+    ::reverse_complement(rev_seq.begin(), rev_seq.end());
+    assert(rev_seq[0] == '\0');
+
+    const auto &alphabet = graph_->alphabet();
+    for (size_t c = 0; c < alphabet.size(); ++c) {
+        if (alphabet[c] != boss::BOSS::kSentinel) {
+            rev_seq[0] = complement(alphabet[c]);
+            node_index next = graph_->kmer_to_node(rev_seq);
+            if (next != DeBruijnGraph::npos)
+                callback(next);
+        }
+    }
+}
+
+bool NodeRC::load(const std::string &filename_base) {
+    const auto rc_filename = utils::make_suffix(filename_base, kRCExtension);
+    try {
+        std::ifstream instream(rc_filename, std::ios::binary);
+        if (!instream.good())
+            return false;
+
+        rc_.load(instream);
+        mapping_.load(instream);
+        return true;
+
+    } catch (...) {
+        logger->error("Cannot load graph RC from file {}", rc_filename);
+        return false;
+    }
+}
+
+void NodeRC::serialize(const std::string &filename_base) const {
+    if (!rc_.size())
+        logger->warn("NodeRC was initialized with set_graph, so nothing to serialize.");
+
+    const auto fname = utils::make_suffix(filename_base, kRCExtension);
+
+    std::ofstream outstream(fname, std::ios::binary);
+    rc_.serialize(outstream);
+    mapping_.serialize(outstream);
+}
+
+bool NodeRC::is_compatible(const SequenceGraph &graph, bool verbose) const {
+    if (!rc_.size()) {
+        if (const auto *dbg = dynamic_cast<const DeBruijnGraph*>(&graph)) {
+            if (dbg == graph_)
+                return true;
+
+            if (verbose)
+                logger->error("Stored graph pointer does not match");
```

Comment:
If it's an error, it has to be reported in every case. So `logger->error` should never be inside `if-verbose`

```suggestion
            logger->trace("Stored graph pointer does not match");
```

Reply:
Also please fix this in other cases below